tc/sgx/trusted_worker_manager/enclave/signup_enclave_wpe.cpp

```diff
@@ -44,6 +45,34 @@ static void CreateSignupReportDataWPE(const uint8_t* ext_data,
     EnclaveData* enclave_data,
     sgx_report_data_t* report_data);
 
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+tcf_err_t ecall_GenerateNonce(uint8_t* out_nonce, size_t in_nonce_size) {
+    tcf_err_t result = TCF_SUCCESS;
+    tcf::error::ThrowIf<tcf::error::ValueError>(in_nonce_size <= 0,
+        "Nonce size should be positive value");
+    try {
+        ByteArray nonce_bytes = tcf::crypto::RandomBitString(in_nonce_size);
+        // Convert nonce to hex string and persist in the EnclaveData
+        EnclaveData* enclaveData = EnclaveData::getInstance();
+        std::string nonce_hex = ByteArrayToHexEncodedString(nonce_bytes);
+        enclaveData->set_nonce(nonce_hex);
```

Comment:
WPE needs to check if this nonce is present in signature obtained from KME (uniqe ID + nonce). Are we doing that ?